### ISSUE TITLE
Sorting the Work-Album list

### ISSUE THREAD

**Issue by darrell-k** [darrell-k/slimserver#5]
See https://forums.slimdevices.com/forum/developer-forums/developers/1655727-adding-classical-work-to-lms-data?p=1681154#post1681154

**Comment by daverz** [comment]
With all the database changes, what's the status of WORKSORT?  There seems to have been a reversion in the numeric sorting.

![Screenshot from 2024-03-23 01-31-35](https://github.com/darrell-k/slimserver/assets/2717422/ece21473-6115-471d-b4ea-71bba59579c9)

**Comment by darrell-k** [comment]
All that change should have done is consolidate 3 database schema update steps into 1,  amounting to the same end result. 

What does your works table look like?

Have you tried a full rescan?

**Comment by daverz** [comment]
Yes, this was a complete re-clone and full rescan.  I guess it's the way I tagged these together with a different sort algorithm.  For a sequence that went up to n digits, I would only provide worksort for the numbers with less than n digits.  I'll have to write a shell script to fix it.

**Comment by darrell-k** [comment]
As long as you are supplying WORKSORT for every track in your database, it should never invoke my n=4 code. Let me know if you find different.

**Comment by daverz** [comment]
In this Haydn example, I don't supply WORKSORT for Symphony No. 100 and above (the out of place No. 94 is a tagging error).  So shouldn't the n=4 algo kick in?  It worked before.

**Comment by darrell-k** [comment]
Yes it should. but if you've supplied a WORKSORT for symphonies < 100 with n<>4, it's going to mess up the sequence.

I'm puzzled that behaviour has changed, though. What does your works table look like in eg sqlitebrowser?

**Comment by daverz** [comment]
Ah, yeah, my worksort for the Haydn symphonies zero pads to 3 places for 1 through 99.  So combined with your padding to 4 places for 100 and above, 0100 lexically sorts before 011.

Maybe I should just rename the WORKSORT tags to see how the default sort works.

What table is the composer_id in?

**Comment by darrell-k** [comment]
Composer is in the contributors table.

**Comment by daverz** [comment]
I ended up removing all the worsort tags and rescanning.  The default sort works fine now.
